codegen/cmd/injection-gen/generators/reconciler_reconciler.go

```diff
@@ -339,13 +339,19 @@ func (r *reconcilerImpl) Reconcile(ctx {{.contextContext|raw}}, key string) erro
 		if reconciler.EventAs(reconcileEvent, &event) {
```

Comment:
Let's follow up by moving the reconciler event logic out of the generated code and into the reconciler base that @whaught is working on.

Reply:
it would be nice to write some tests on this area of the codebase.

Reply:
Good idea, I'll see how I can help.